### DISCUSSION TITLE
Move some refresh based test into integration

### DISCUSSION THREAD

**Discussion by WebReflection** [pyscript/pyscript#2233] [Proposals]
There are test use-cases that require manual intervention mostly to be sure the browser's cache worked, or even the IndexedDB we use behind the scene, and it'd be awesome to use a *new tab* approach, as suggested [in here](https://github.com/pyscript/pyscript/issues/2228#issuecomment-2441384753).

We should take some time to have that ability in both here and *polyscript* project so that we can be sure those tests are also automatically covered without requiring ourselves to manually test things "*just work*".

**Comment by Archmonger** [comment]
As a note, Playwright also supports forcibly refreshing an existing tab via the `Page` context.

**Reply by WebReflection** [reply]
by "*forcibly refreshing*" do you mean as `window.reload(true)` where also the cache gets busted? Although that shouldn't affect the IndexedDB state, I will explore this possibility.

**Reply by Archmonger** [reply]
Not all in one command, but there's probably a workaround for clearing all browser data.

Reload page API: https://playwright.dev/python/docs/api/class-page#page-reload

I'm assuming if you destruct the entire [browser context](https://playwright.dev/python/docs/api/class-browsercontext) and create a new one then it's a "factory reset" browser.

Additionally you can [evaluate](https://playwright.dev/python/docs/api/class-page#page-evaluate) JavaScript on demand during your tests, so `window.reload(true)` can be executed.

**Reply by WebReflection** [reply]
Yeah, we are using `eval` and indeed that could've been an option but I think in here I've tackled the IndexedDB story with a pinned version of a module and I can reload and check that indeed on a second run *micropip* logs are gone https://github.com/pyscript/pyscript/pull/2234

There was another test for `lockFileURL` already but this one ensures pinned versions too, which I think was the desirable goal.

Thanks for the details though, appreciated.
